src/theory/bags/card_solver.cpp

```diff
@@ -331,12 +331,21 @@ void CardSolver::checkLeafBag(const std::pair<Node, Node>& pair,
         bags::InferInfo sumInfo(&d_im, InferenceId::BAGS_CARD);
         Node sumLEQ = d_nm->mkNode(LEQ, sum, pair.second);
         sumInfo.d_conclusion = premise.negate().orNode(sumLEQ);
-        d_im.lemmaTheoryInference(&sumInfo);
+        sendInferInfo(sumInfo);
       }
     }
   }
 }
 
+void CardSolver::sendInferInfo(InferInfo& inferInfo)
+{
+  if (!inferInfo.isCachedLemma())
+  {
+    d_hasNewPendingLemmas = true;
```

Comment:
I don't think this is accurate.  You are actually sending a lemma here, not adding it to pending.

Reply:
Done. Actually I hated the new boolean.